src/components/ApiPublishModal/OpenApiSpec.tsx

```diff
@@ -57,14 +79,84 @@ const OpenApiSpec: React.FC<IProps> = props => {
           },
         ]}
       >
-        <S.Textarea rows={15} placeholder="Enter OpenAPI Spec in YAML/JSON format" />
+        <S.Textarea
+          rows={15}
+          placeholder="Enter OpenAPI Spec in YAML/JSON format"
+          onChange={e => {
+            const spec = e.target.value;
+
+            if (!spec) {
+              setWarnings([]);
+            } else {
+              setWarnings(checkMockingExamples(YAML.parse(JSON.parse(JSON.stringify(spec)))));
+            }
+          }}
+        />
       </Form.Item>
 
+      {renderedWarnings}
+
       <Form.Item name={['mocking', 'enabled']} valuePropName="checked">
         <Checkbox onChange={e => setIsApiMocked(e.target.checked)}>Enable mocking</Checkbox>
       </Form.Item>
     </>
   );
 };
 
+const findResponseExample = (key: string, children: any, check: {hasExample: boolean}) => {
+  if ((key === 'example' && children) || (key === 'examples' && children && Object.entries(children).length)) {
```

Comment:
will this ever check for the second part after `||` ?

maybe you can do this instead:
`if (key === 'examples' && typeof children === 'object' && Object.entries(children).length)`

Reply:
It will still check after `||` as the keys are different ( `example` and `examples` )